data_structures/src/builders.rs

```diff
@@ -0,0 +1,133 @@
+extern crate rand;
+
+use std::convert::TryFrom;
+use std::net::SocketAddr;
+use std::time::{SystemTime, UNIX_EPOCH};
+use std::u32::MAX as U32_MAX;
+
+use rand::{thread_rng, Rng};
+
+use crate::types::{Address, Command, IpAddress, Message};
+
+////////////////////////////////////////////////////////////////////////////////////////
+// PROTOCOL MESSAGES CONSTANTS
+////////////////////////////////////////////////////////////////////////////////////////
+/// Magic number
+pub const MAGIC: u16 = 0xABCD;
+
+/// Version
+pub const VERSION: u32 = 0x0000_0001;
+
+/// Capabilities
+pub const CAPABILITIES: u64 = 0x0000_0000_0000_0001;
+
+/// User agent
+pub const USER_AGENT: &str = "full-node-desktop-edition";
+
+/// Genesis block
+pub const GENESIS: u64 = 0x0123_4567_89AB_CDEF;
+
+////////////////////////////////////////////////////////////////////////////////////////
+// BUILDER PUBLIC FUNCTIONS
```

Comment:
I see...these builder generate a `Message` with `Command`, so why not `impl Message {}` instead of setting them free.